website/client/src/pages/settings/settingRows/fixValuesSetting.vue

```diff
@@ -0,0 +1,295 @@
+    margin: 0;
+  }
+}
+
+.svg-icon.level {
+  color: $gray-200;
+
+  :global svg path {
+    fill: currentColor;
+  }
+}
+
+</style>
+
+<script>
+// import clone from 'lodash/clone';
+import SaveCancelButtons from '../components/saveCancelButtons.vue';
+import { InlineSettingMixin } from '../components/inlineSettingMixin';
+import healthIcon from '@/assets/svg/health.svg';
+import experienceIcon from '@/assets/svg/experience.svg';
+import manaIcon from '@/assets/svg/mana.svg';
+import svgGold from '@/assets/svg/gold.svg';
+import level from '@/assets/svg/level.svg';
+import streakIcon from '@/assets/svg/streak.svg';
+import { mapState } from '@/libs/store';
+import { MAX_LEVEL_HARD_CAP } from '../../../../../common/script/constants';
+
+export default {
+  components: { SaveCancelButtons },
+  mixins: [InlineSettingMixin],
+  data () {
+    return {
+      restoreValues: {
+        hp: 0,
+        mp: 0,
+        gp: 0,
+        exp: 0,
+        lvl: 0,
+        streak: 0,
+      },
+      icons: Object.freeze({
+        health: healthIcon,
+        experience: experienceIcon,
+        mana: manaIcon,
+        gold: svgGold,
+        level,
+        streak: streakIcon,
+      }),
+      inputList: Object.freeze([
+        {
+          translationKey: 'health',
+          icon: healthIcon,
+          property: 'hp',
+        },
+        {
+          translationKey: 'experience',
+          icon: experienceIcon,
+          property: 'exp',
+        }, {
+          translationKey: 'mana',
+          icon: manaIcon,
+          property: 'mp',
+        }, {
+          translationKey: 'gold',
+          icon: svgGold,
+          property: 'gp',
+        },
+        {
+          translationKey: 'level',
+          icon: level,
+          property: 'lvl',
+        },
+        {
+          translationKey: 'fix21Streaks',
+          icon: streakIcon,
+          property: 'streak',
+        },
+      ]),
+    };
+  },
+  computed: {
+    ...mapState({ user: 'user.data' }),
+  },
+  mounted () {
+    this.resetControls();
+  },
+  methods: {
+    resetControls () {
+      const {
+        hp, mp, gp, exp, lvl,
+      } = this.user.stats;
+
+      this.restoreValues = {
+        hp, mp, gp, exp, lvl, streak: this.user.achievements.streak,
+      };
+    },
+    close () {
+      this.validateInputs();
+    },
+    markAsChanged (inputType, keyupEvent) {
+      this.restoreValues[inputType.property] = keyupEvent.target.value;
+      this.modalValuesChanged();
+    },
+    save () {
+      if (!this.validateInputs()) {
+        return;
+      }
+
+      if (this.restoreValues.lvl > MAX_LEVEL_HARD_CAP) {
+        this.restoreValues.lvl = MAX_LEVEL_HARD_CAP;
+      }
+
+      const userChangedLevel = this.restoreValues.lvl !== this.user.stats.lvl;
+      const userDidNotChangeExp = this.restoreValues.exp === this.user.stats.exp;
+      if (userChangedLevel && userDidNotChangeExp) {
+        this.restoreValues.exp = 0;
+      }
+
+      // todo
```

Comment:
todo?

Reply:
seems I already had it working with the lines below the todo - will be removed